### PR TITLE
feat(app-cmds): add support for 3.10 unions

### PR DESCRIPTION
**Pull request by alentoghostflame**
Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

## Summary

Adds PEP 604 support to application commands, allowing the Python 3.10 style of unions.
Example: `arg1: str | None`

## Checklist

- [X] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/application_command.py [hunk 1]
```diff
@@ -92,6 +93,13 @@
 else:
     _CustomTypingMetaBase = object
 
+if sys.version_info[1] >= 10:
```
Comment (DenverCoder1) [inline nextcord/application_command.py:96]: This is more specific and readable. Also supported by conditional Pyright checks I believe (though I'm not 100% sure).

```suggestion
if sys.version_info >= (3, 10):
```

Comment (alentoghostflame) [reply nextcord/application_command.py]: Tested on 3.10 and 3.8 and commited.

**Review by EmreTech — APPROVED** [review]
Haven't tested
Although, I think the title should be changed to explicitly state that this adds support for 3.10 unions because giving the pep number doesn't showcase this right away.

**Review by DenverCoder1 — APPROVED** [review]
Haven't tested but looks good to me.

File: nextcord/application_command.py [hunk 2]
```diff
@@ -92,6 +93,13 @@
 else:
     _CustomTypingMetaBase = object
 
+if sys.version_info >= (3, 10):
```
Comment (DenverCoder1) [inline nextcord/application_command.py:96]: I think `utils.PY_310` can be used for this. I just came across it and it looks like it's exactly this condition but in a constant.

**Review by teaishealthy — APPROVED** [review]

**Review by ooliver1 — APPROVED** [review]
Tested with channel type unions.

**Pull request merged**